### PR TITLE
Log debug when using generic message types

### PR DESCRIPTION
**Pull request by PhilBastian**
as per [the documentation](https://docs.particular.net/nservicebus/messaging/messages-events-commands#designing-messages), generic message types are not supported. This change adds a debug log to that effect

### CODE REVIEW

**Comment by PhilBastian** [comment]
I have raised as a draft PR so that it isn't merged while the historical question is considered internally

File: src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs [hunk 1]
```diff
@@ -165,15 +164,18 @@ static IEnumerable<Type> GetHandledMessageTypes(Type messageHandlerType)
 
     MessageMetadata RegisterMessageType(Type messageType)
     {
+        if (messageType.IsGenericType)
+        {
+            // This is not an error because in most cases it will work, but it's still not supported should issues arise
+            Logger.Warn($"Generic messages types are not supported. Consider converting {messageType.AssemblyQualifiedName} to a dedicated, simple type");
```
Comment (danielmarbach) [inline src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs:170]: How about we do log debug only?

Comment (danielmarbach) [inline src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs:170]:
```suggestion
            Logger.Warn($"Generic messages types are not supported. Consider converting '{messageType.AssemblyQualifiedName}' to a dedicated, simple type");
```

Comment (danielmarbach) [reply src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs:170]: Not super necessary but would it be possible with little effort to write a unit test?

Comment (danielmarbach) [reply src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs:170]: We have decided not to write a test but switch the level to debug only to hint to developers that there are better ways to design their messages.

**Review by danielmarbach — APPROVED** [review]

**Comment by WilliamBZA** [comment]
Are you  happy with the new name for the PR? `Add a debug Log entry when using generic message types`?

Does this need to be back-ported?

**Comment by PhilBastian** [comment]
> Are you happy with the new name for the PR? `Add a debug Log entry when using generic message types`?
> 
> Does this need to be back-ported?

I wouldn't think it needs to be backported, since it's just a hint to a developer that they are not designing the messaging model 'correctly' and no functionality is affected.

The title says what it does...

**Pull request merged**